scale/ingest/test/test_views.py

```diff
@@ -912,6 +927,52 @@ def test_ingest_after_dry_run(self, mock_msg_mgr):
         self.assertTrue(isinstance(result, dict), 'result  must be a dictionary')
         self.assertIsNotNone(result['job'])
 
+    @patch('ingest.models.CommandMessageManager')
+    @patch('ingest.models.create_cancel_jobs_messages')
+    def test_cancel_scan_job(self, msg_create, mock_msg_mgr):
+        """Tests successfully calling the Scan process view for an ingest Scan. following dry run"""
+
+        self.scan.job = job_utils.create_job()
+        self.scan.save()
+        self.ingest = ingest_test_utils.create_ingest(scan=self.scan, file_name='test3.txt',
+                                                       status='QUEUED')
+
+        url = '/%s/scans/cancel/%d/' % (self.api, self.scan.id)
+        response = self.client.generic('POST', url, json.dumps({ 'ingest': True }), 'application/json')
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED, response.content)
+        
+        result = json.loads(response.data)
+
+        msg_create.assert_called()
+        mock_msg_mgr.assert_called()
+
+        self.assertEqual(len(result), 2)
+
+    @patch('ingest.models.CommandMessageManager')
+    @patch('ingest.models.create_cancel_jobs_messages')
+    def test_cancel_scan_job_nothing_to_cancel(self, msg_create, mock_msg_mgr):
+        """Tests no cancel messages generated when jobs are not in a cancelable state"""
+
+        self.scan.job = job_utils.create_job()
+        self.scan.job.status = "COMPLETED"
+        self.scan.job.save()
+        self.scan.save()
+        self.ingest = ingest_test_utils.create_ingest(scan=self.scan, file_name='test3.txt',
```

Comment:
Any reason to not set the job status as CANCELED here?

Reply:
both completed and canceled are not 'cancelable' states, so I'm testing both at the same time here.